src/jit/codegencommon.cpp

```diff
@@ -4361,16 +4364,22 @@ void            CodeGen::genFnPrologCalleeRegArgs(regNumber xtraReg,
         {
             storeType = TYP_I_IMPL;   // Default store type for a struct type is a pointer sized integer
 #if FEATURE_MULTIREG_ARGS
-            // Must be <= 32 bytes or else it wouldn't be passed in registers
+            // Must be <= MAX_PASS_MULTIREG_BYTES or else it wouldn't be passed in registers
             noway_assert(varDsc->lvSize() <= MAX_PASS_MULTIREG_BYTES);
 #endif // FEATURE_MULTIREG_ARGS
 #ifdef FEATURE_UNIX_AMD64_STRUCT_PASSING
             storeType = regArgTab[argNum].type;
 #endif // !FEATURE_UNIX_AMD64_STRUCT_PASSING
             if (varDsc->lvIsHfaRegArg())
             {
+#ifdef _TARGET_ARM_
+                // On ARM32 the storeType for HFA args is always TYP_FLOAT
+                storeType = TYP_FLOAT;
+                slotSize = (unsigned)emitActualTypeSize(storeType);
+#else   // _TARGET_ARM64_
                 storeType = genActualType(varDsc->GetHfaType());
-                slotSize  = (unsigned) emitActualTypeSize(storeType);
+                slotSize = (unsigned)emitActualTypeSize(storeType);
+#endif // _TARGET_ARM64_
             }
```

Comment:
When I tested only with these changes, there was another assertion failure. Here is the assertion failure that I encountered.
```
Assert failure(PID 24305 [0x00005ef1], Thread: 24305 [0x5ef1]): Assertion failed 'regArgMaskLiveSave != regArgMaskLive' in 'DomainBoundILStubClass:IL_STUB_PInvoke(struct):struct' (IL size 40)

    File: /home/parjong/projects/toolchains/coreclr/src/jit/codegencommon.cpp Line: 4912
    Image: /mnt/dotnet-latest/Linux.arm.Debug/corerun

**** MessageBox invoked, title 'corerun - Assert Failure (PID 24305, Thread 24305/0x5ef1)' ****
  Assertion failed 'regArgMaskLiveSave != regArgMaskLive' in 'DomainBoundILStubClass:IL_STUB_PInvoke(struct):struct' (IL size 40)


/home/parjong/projects/toolchains/coreclr/src/jit/codegencommon.cpp, Line: 4912

Abort - Kill program
Retry - Debug
Ignore - Keep running
```
It seems that ``lvHfaSlots`` need to be revised in order to fix #5326.